push-notifications.js

```diff
@@ -27,15 +32,60 @@ function PushNotifications(options) {
     this.secretKey = options.secretKey;
 
     if (!options.hasOwnProperty('endpoint')) {
+        this.protocol = 'https';
         this.endpoint = `${this.instanceId}.pushnotifications.pusher.com`;
     } else if (typeof options.endpoint !== 'string') {
         throw new Error('endpoint must be a string');
     } else {
+        this.protocol = 'http';
         this.endpoint = options.endpoint;
     }
+
+    baseRequest = request.defaults({
+        headers: {
+            Authorization: `Bearer ${this.secretKey}`,
```

Comment:
What happens if I have multiple instances of this class with different secret keys?